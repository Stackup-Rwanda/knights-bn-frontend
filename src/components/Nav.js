import React from 'react';
import '../assets/styles/styles.scss';
import { Link } from 'react-router-dom';
import logo from '../assets/images/logo@2x.png';

const Nav = () => (
  <div className="nav ">
    <Link to="/">
      <h3 className="text-center pt-2 li">
        <img className="logo" src={logo} />
      </h3>
    </Link>
    <ul className="nav-link">
      <Link to="/login">
        <li className="li">Login</li>
      </Link>
      <Link to="/">
        <li className="li">SignUp</li>
      </Link>
    </ul>
  </div>
);

export default Nav;
