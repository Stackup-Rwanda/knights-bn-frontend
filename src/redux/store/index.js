import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
// import logger from 'redux-logger';
import rootReduce from '../reducers/index';

const store = createStore(
  rootReduce,
  applyMiddleware(thunk),
);

export default store;
