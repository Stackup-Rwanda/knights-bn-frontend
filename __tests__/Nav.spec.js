import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Nav from '../src/components/Nav';
import store from '../src/redux/store';

const NavComponent = () => render(
  <Provider store={store}>
    <Router>
      <Nav />
    </Router>
  </Provider>,
);
describe('=============>Nav<=============', () => {
  afterEach(cleanup);
  it('should render Nav component', () => {
    const { asFragment } = NavComponent();
    expect(asFragment(<Nav />)).toMatchSnapshot();
  });
});
