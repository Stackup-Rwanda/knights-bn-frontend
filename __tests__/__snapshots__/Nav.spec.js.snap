// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`=============>Nav<============= should render Nav component 1`] = `
<DocumentFragment>
  <div
    class="nav "
  >
    <a
      href="/"
    >
      <h3
        class="text-center pt-2 li"
      >
        <img
          class="logo"
          src=""
        />
      </h3>
    </a>
    <ul
      class="nav-link"
    >
      <a
        href="/login"
      >
        <li
          class="li"
        >
          Login
        </li>
      </a>
      <a
        href="/"
      >
        <li
          class="li"
        >
          SignUp
        </li>
      </a>
    </ul>
  </div>
</DocumentFragment>
`;
